### PR TITLE
Capability validation API

### PR DESCRIPTION
**Pull request by HenryNguyen5**
Slimmed this down to just the validation struct, will add the integration into capabilities into a diff PR

### CODE REVIEW

File: pkg/capabilities/capabilities.go [hunk 1]
```diff
@@ -105,6 +105,28 @@ type CallbackExecutable interface {
...
+	//  - `RegisterTrigger` and `UnregisterTrigger`
+	//
+	// They take in a request object that has a `Config` field that may conflate
+	// with the `Config` field within `Execute`.
+	GetRequestConfigJsonSchema() (*CapabilityResponse)
+	// ValidateRequestConfig(config *values.Map) error
+	// ValidateRequestInput(inputs *values.Map) error
+	// NOTE: Should we need this? Or should we just document this within
+	// the capability itself?
+	// ExampleOutput() *values.Map
```
Comment (DeividasK) [inline pkg/capabilities/capabilities.go:127]: We need this to validate input/output compatibility. We should be able to do something like
```
SrcCapability.Validate(DestCapability.ExampleOutput())
```

The above can catch some basic mismatches between something like
```
consensus:
  - type: offchain_reporting@1
    ref: evm_median
    inputs:
      observations:
        - triggers.report_data.outputs
    config:
      aggregation_method: data_feeds_2_0
      aggregation_config:
        123: # ETHUSD
          deviation: "0.005"
          heartbeat: 24h
        456: # LINKUSD
          deviation: "0.001"
          heartbeat: 24h
        789: # USDBTC
          deviation: "0.002"
          heartbeat: 6h
      encoder: EVM
      encoder_config:
          abi: "mercury_reports bytes[]"

targets:
  - type: write_polygon_mainnet@1
    inputs:
      report:
        - consensus.evm_median.outputs # SHOULD BE consensus.evm_median.outputs.report
    config:
      address: "0xaabbcc"
      method: "updateFeedValues(report bytes, role uint8)"
      params: [$(inputs.report), 1] # THIS IS AN ERROR, BECAUSE REPORT IS report.report
```

File: pkg/capabilities/capabilities.go [hunk 2]
```diff
@@ -105,6 +105,28 @@ type CallbackExecutable interface {
 	Execute(ctx context.Context, callback chan<- CapabilityResponse, request CapabilityRequest) error
 }
 
+type Validatable interface {
```
Comment (DeividasK) [inline pkg/capabilities/capabilities.go:108]: Could this be on `BaseCapability`?

File: pkg/capabilities/capabilities.go [hunk 3]
```diff
@@ -105,6 +105,28 @@ type CallbackExecutable interface {
...
+	// the config field against the schema.
+	//
+	// NOTE: What to do with method calls like:
+	//
+	//  - `RegisterToWorkflow` and `UnregisterFromWorkflow` 
+	//  - `RegisterTrigger` and `UnregisterTrigger`
+	//
+	// They take in a request object that has a `Config` field that may conflate
+	// with the `Config` field within `Execute`.
+	GetRequestConfigJsonSchema() (*CapabilityResponse)
```
Comment (DeividasK) [inline pkg/capabilities/capabilities.go:122]: I expected
```suggestion
	ValidateConfig() (any, error)
```

Not sure if any can be made into a generic that returns the capability-specific interface.

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 4]
```diff
@@ -92,10 +93,10 @@ func (o *capability) HealthReport() map[string]error {
 }
 
 type workflowConfig struct {
-	AggregationMethod string      `mapstructure:"aggregation_method"`
-	AggregationConfig *values.Map `mapstructure:"aggregation_config"`
-	Encoder           string      `mapstructure:"encoder"`
-	EncoderConfig     *values.Map `mapstructure:"encoder_config"`
+	AggregationMethod string      `mapstructure:"aggregation_method" json:"aggregation_method,omitempty"`
```
Comment (DeividasK) [inline pkg/capabilities/consensus/ocr3/capability.go:96]: Why `omitempty`?

File: pkg/capabilities/pb/capabilities.proto [hunk 5]
```diff
@@ -49,6 +49,7 @@ message UnregisterTriggerRequest {
 service TriggerExecutable {
   rpc RegisterTrigger (RegisterTriggerRequest) returns (google.protobuf.Empty) {}
   rpc UnregisterTrigger (UnregisterTriggerRequest) returns (google.protobuf.Empty) {}
+  rpc GetRequestConfigJsonSchema(google.protobuf.Empty) returns (CapabilityResponse) {}
```
Comment (DeividasK) [inline pkg/capabilities/pb/capabilities.proto:52]: Hmmm, I was thinking that this would be something that's internal to the capability, i.e., when `RegisterTrigger` is called, it (internally) calls `ValidateConfig` and returns an error if config is invalid, otherwise it stores the CapabilitySpecific config internally.

File: pkg/capabilities/capabilities.go (hunk 2 above)
Comment (HenryNguyen5) [reply pkg/capabilities/capabilities.go:108]: Yes it should be on `BaseCapability`! But I was making incremental changes first. I didn't want to make a change across all capability types at the same time.

File: pkg/capabilities/consensus/ocr3/capability.go (hunk 4 above)
Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability.go:96]: No particular reason, I can remove it

File: pkg/capabilities/pb/capabilities.proto (hunk 5 above)
Comment (HenryNguyen5) [reply pkg/capabilities/pb/capabilities.proto:52]: I was originally thinking what you are proposing here too, but what I realized is that by exposing this via endpoint we can provide a much better DX.

With this endpoint, we can do something like creating a full JSON schema that can be used to lint a YAML workflow file.

Comment (HenryNguyen5) [reply pkg/capabilities/pb/capabilities.proto:52]: I also want to internally call `ValidateConfig` during entrypoints like `RegisterTrigger`, then we get both:
1. The ability to provide workflow authors with a JSON schema meant for linting
2. Capabilities always validate their config and inputs upon execution

File: pkg/capabilities/capabilities.go (hunk 3 above)
Comment (HenryNguyen5) [reply pkg/capabilities/capabilities.go:122]: I was thinking of exposing another endpoint called `ValidateConfig` that returns nothing or a list of errors associated with the config.

File: pkg/capabilities/capabilities.go (hunk 1 above)
Comment (HenryNguyen5) [reply pkg/capabilities/capabilities.go:127]: Do we expect this to be a common use case? For some reason I thought that was an exception rather than the norm. I was thinking that often when we're chaining capabilities we might be passing in some other inputs too.

File: pkg/capabilities/consensus/ocr3/capability_test.go [hunk 6]
```diff
@@ -20,6 +23,15 @@ import (
 const workflowTestID = "consensus-workflow-test-id-1"
 const workflowExecutionTestID = "consensus-workflow-execution-test-id-1"
 
+var transformJSON = cmp.FilterValues(func(x, y []byte) bool {
```
Comment (HenryNguyen5) [inline pkg/capabilities/consensus/ocr3/capability_test.go:26]: I ended up copy / pasting this multiple times... where should i put this if i want to have test helpers?

File: pkg/capabilities/capabilities.go [hunk 7]
```diff
@@ -105,10 +105,33 @@ type CallbackExecutable interface {
...
+	// the config field against the schema.
+	//
+	// NOTE: What to do with method calls like:
+	//
+	//  - `RegisterToWorkflow` and `UnregisterFromWorkflow`
+	//  - `RegisterTrigger` and `UnregisterTrigger`
+	//
+	// They take in a request object that has a `Config` field that may conflate
+	// with the `Config` field within `Execute`.
+	GetRequestConfigJSONSchema() *CapabilityResponse
```
Comment (cedric-cordenier) [inline pkg/capabilities/capabilities.go:122]: @HenryNguyen5 No objection to using JSONSchema for validation, I think that's great, but I'm a bit concerned about the API here.

Since this is truly generic (i.e. all capabilities) will provide us with a JSONSchema to validate against, I think we can be very concrete with our API and not use something like a CapabilityResponse. Some concrete JSONSchema type would be fine 👍

File: pkg/values/map.go [hunk 8]
```diff
@@ -12,6 +13,12 @@ type Map struct {
 	Underlying map[string]Value
 }
 
+func (m Map) JSONSchema() *jsonschema.Schema {
```
Comment (cedric-cordenier) [inline pkg/values/map.go:16]: @HenryNguyen5 Why is this needed?

File: pkg/capabilities/capabilities.go (hunk 7 above)
Comment (cedric-cordenier) [reply pkg/capabilities/capabilities.go:122]: On your comment about multiple method calls, I think having a single schema per call is fine 👍  or alternatively having a single method which returns a map of method names to schemas.

File: pkg/capabilities/consensus/ocr3/capability_test.go (hunk 6 above)
Comment (cedric-cordenier) [reply pkg/capabilities/consensus/ocr3/capability_test.go:26]: If these are test helpers my preference is to leave them copied since I don't think it harms much.

However there is a test utils package in common IIRC, so you could put it there too.

File: pkg/capabilities/capabilities.go (hunk 7 above)
Comment (cedric-cordenier) [reply pkg/capabilities/capabilities.go:122]: Ah, the name feels pretty clunky too!

Getters in golang usually omit the GetX prefix, so would rename this to InputsSchema(), and maybe add a ConfigSchema() for config fields.

File: pkg/capabilities/triggers/mercury_trigger.go [hunk 9]
```diff
@@ -26,6 +26,7 @@ var mercuryInfo = capabilities.MustNewCapabilityInfo(
 // This Trigger Service allows for the registration and deregistration of triggers. You can also send reports to the service.
 type MercuryTriggerService struct {
 	capabilities.CapabilityInfo
+	*capabilities.Validator
```
Comment (cedric-cordenier) [inline pkg/capabilities/triggers/mercury_trigger.go:29]: Nit: if you make Validator a value type and add generics to it, then you can one line this addition into a fairly neat:

```
capabilities.Validator[mercury.FeedReport]
```

with no instantiation required.

Comment (HenryNguyen5) [reply pkg/capabilities/triggers/mercury_trigger.go:29]: whoa, where can i read about this distinction / use case?

File: pkg/capabilities/consensus/ocr3/capability_test.go (hunk 6 above)
Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability_test.go:26]: yeah they're test helpers. Okay, I'll keep copy pasting them for now then.

File: pkg/capabilities/capabilities.go (hunk 2 above)
Comment (HenryNguyen5) [reply pkg/capabilities/capabilities.go:108]: Added.

File: go.mod [hunk 10]
```diff
@@ -39,6 +39,13 @@ require (
...
 )
 
+require (
+	github.com/bahlo/generic-list-go v0.2.0 // indirect
+	github.com/buger/jsonparser v1.1.1 // indirect
+	github.com/mailru/easyjson v0.7.7 // indirect
+	github.com/wk8/go-ordered-map/v2 v2.1.8 // indirect
+)
+
 require (
```
Comment (jmank88) [inline go.mod:49]: Please combine and tidy these `indirect` blocks:
```suggestion
```

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 11]
```diff
@@ -92,10 +97,10 @@ func (o *capability) HealthReport() map[string]error {
 }
 
 type workflowConfig struct {
-	AggregationMethod string      `mapstructure:"aggregation_method"`
```
Comment (jmank88) [inline pkg/capabilities/consensus/ocr3/capability.go:95]: Are we losing portability by using irregular struct tags like `jsonschema`?

Does the experimental v2 API meet our needs?
- https://pkg.go.dev/github.com/go-json-experiment/json
- https://github.com/golang/go/discussions/63397

Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability.go:95]: The `jsonschema` struct tag doesn't affect json (de)serialization, it instructs the jsonschema package on how to generate a JSON schema. Unfortunately that new API doesn't help us here

Comment (jmank88) [reply pkg/capabilities/consensus/ocr3/capability.go:95]: Are the schemas a standard format, or specific to `jsonschema`?

Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability.go:95]: They are a standard https://json-schema.org/

File: pkg/capabilities/triggers/mercury_trigger.go [hunk 12]
```diff
@@ -24,8 +23,44 @@ var mercuryInfo = capabilities.MustNewCapabilityInfo(
 
 const defaultTickerResolutionMs = 1000
 
+type MercuryTriggerOutput struct {
```
Comment (bolekk) [inline pkg/capabilities/triggers/mercury_trigger.go:26]: This is the same as TriggerEvent, which is a generic struct to support all triggers, not only Mercury. Why do we need a separate one here?

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 13]
```diff
@@ -29,9 +28,17 @@ var info = capabilities.MustNewCapabilityInfo(
 	"v1.0.0",
 )
 
+type CapabilityConfig = workflowConfig
```
Comment (cedric-cordenier) [inline pkg/capabilities/consensus/ocr3/capability.go:31]: No need for the type alias here I think: just rename `workflowConfig` to `CapabilityConfig`

I think you can simplify the naming btw: CapabilityConfig -> Config; CapabilityInputs -> Inputs, etc

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 14]
```diff
@@ -50,10 +57,13 @@ type capability struct {
 }
 
 var _ capabilityIface = (*capability)(nil)
+var _ capabilities.ConsensusCapability = (*capability)(nil)
```
Comment (cedric-cordenier) [inline pkg/capabilities/consensus/ocr3/capability.go:60]: @HenryNguyen5 How come you added this? I think this should do the same job as capabilityIface (my intent was to loosely couple this to the consensus interface so we only pulled in what we needed)

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 15]
```diff
@@ -50,10 +57,13 @@ type capability struct {
...
+var OCR3CapabilityValidator = capabilities.NewValidator[CapabilityConfig, CapabilityInputs, CapabilityOutputs](capabilities.ValidatorArgs{Info: info})
```
Comment (cedric-cordenier) [inline pkg/capabilities/consensus/ocr3/capability.go:61]: Hmmmm... does this need to be public, and do we need to instantiate at the package level (i.e. can we just move it into newCapability)?

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 16]
```diff
@@ -92,22 +102,14 @@ func (o *capability) HealthReport() map[string]error {
...
-	AggregationMethod string      `mapstructure:"aggregation_method"`
-	AggregationConfig *values.Map `mapstructure:"aggregation_config"`
-	Encoder           string      `mapstructure:"encoder"`
-	EncoderConfig     *values.Map `mapstructure:"encoder_config"`
-}
-
-func newWorkflowConfig() *workflowConfig {
-	return &workflowConfig{
-		EncoderConfig:     values.EmptyMap(),
-		AggregationConfig: values.EmptyMap(),
```
Comment (cedric-cordenier) [inline pkg/capabilities/consensus/ocr3/capability.go:104]: @HenryNguyen5 Did anything break when you removed this? IIRC I added this to ensure the map was properly initialized (with its internal map[string]any)

File: pkg/capabilities/consensus/ocr3/capability_test.go [hunk 17]
```diff
@@ -136,7 +175,12 @@ func TestOCR3Capability_Registration(t *testing.T) {
 	cp := newCapability(s, fc, 1*time.Second, mockEncoderFactory, lggr)
 	require.NoError(t, cp.Start(ctx))
 
-	config, err := values.NewMap(map[string]any{"aggregation_method": "data_feeds_2_0"})
+	config, err := values.NewMap(map[string]any{
+		"aggregation_method": "data_feeds_2_0",
+		"aggregation_config": map[string]any{},
```
Comment (cedric-cordenier) [inline pkg/capabilities/consensus/ocr3/capability_test.go:180]: @HenryNguyen5 This is what caused me to add a values.EmptyMap -- is this guaranteed to be present because of the schema validation? (I assume so)

File: pkg/capabilities/generate_schemas/main.go [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+
+import (
+	"fmt"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/consensus/ocr3"
+	"github.com/smartcontractkit/chainlink-common/pkg/capabilities/triggers"
+)
+
+func main() {
```
Comment (cedric-cordenier) [inline pkg/capabilities/generate_schemas/main.go:11]: Since this is a `cmd` can we put it in `cmd/capabilities/generate_schemas/main.go` or similar?

File: pkg/capabilities/pb/capabilities.proto [hunk 19]
```diff
@@ -26,6 +26,7 @@ message CapabilityInfoReply {
   string version = 4;
 }
 
+
```
Comment (cedric-cordenier) [inline pkg/capabilities/pb/capabilities.proto:29]: Extra line :)

File: pkg/capabilities/triggers/mercury_trigger_test.go [hunk 20]
```diff
@@ -13,43 +16,56 @@ import (
...
-	feedTwo   = "0x2222222222222222222200000000000000000000000000000000000000000000"
-	feedThree = "0x3333333333333333333300000000000000000000000000000000000000000000"
-	feedFour  = "0x4444444444444444444400000000000000000000000000000000000000000000"
-	feedFive  = "0x5555555555555555555500000000000000000000000000000000000000000000"
-)
-
-func TestMercuryTrigger(t *testing.T) {
-	ts := NewMercuryTriggerService(100, logger.Nop())
-	ctx := tests.Context(t)
+func registeredTriggerHelper(
```
Comment (cedric-cordenier) [inline pkg/capabilities/triggers/mercury_trigger_test.go:19]: @HenryNguyen5 Can we call this something more descriptive that says what it does, eg. registerTrigger?

File: pkg/capabilities/validate.go [hunk 21]
```diff
@@ -0,0 +1,210 @@
...
+	}
+
+	return Validator[Config, Input, Output]{
+		ValidatorArgs: args,
+		schemas:       make(map[string]string),
+	}
+}
+
+func (v *Validator[Config, Input, Output]) Schema() (string, error) {
+	type Combined struct {
```
Comment (cedric-cordenier) [inline pkg/capabilities/validate.go:77]: This (and CombinedWithoutInput below) can be private since they're declared inside a function.

File: pkg/capabilities/validate.go [hunk 22]
```diff
@@ -0,0 +1,210 @@
...
+
+	schemas map[string]string
+}
+
+var uriPrefix = "https://github.com/smartcontractkit/chainlink/capabilities/"
+
+var _ Validatable = (*Validator[any, any, any])(nil)
+
+type ValidatorArgs struct {
+	Info CapabilityInfo
```
Comment (cedric-cordenier) [inline pkg/capabilities/validate.go:30]: @HenryNguyen5 It would be nice to not depend on CapabilityInfo if we can here. It looks like we only need it for the ID, is that correct? If so, maybe we can just pass in the ID?

File: pkg/capabilities/validate.go [hunk 23]
```diff
@@ -0,0 +1,210 @@
...
+		return nil, err
+	}
+
+	err = jsonSchema.Validate(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	var decodedValue *DecodedValue
+	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
```
Comment (cedric-cordenier) [inline pkg/capabilities/validate.go:160]: If we're decoding `value` into a struct here, we should try to use UnwrapTo if we can -- same thing with the custom decode hook: if we're not handling an edge case correctly, let's put it in the values library rather than here.

File: pkg/capabilities/validate.go [hunk 24]
```diff
@@ -0,0 +1,210 @@
...
+	})
+	if err != nil {
+		return nil, err
+	}
+	err = decoder.Decode(raw)
+	return decodedValue, err
+}
+
+func schemaWith(reflector *jsonschema.Reflector, schemaType any, schemaCache map[string]string, key string, info CapabilityInfo) (string, error) {
+	if reflector == nil {
```
Comment (cedric-cordenier) [inline pkg/capabilities/validate.go:172]: @HenryNguyen5 Can you avoid this by making reflector a `jsonschema.Reflector` to guarantee it's not nil.

File: pkg/capabilities/consensus/ocr3/capability.go (hunk 16 above)
Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability.go:104]: Surprisingly not? Unless somehow it'd not caught by the test suite we currently have. I made sure all tests were passing as I was refactoring

File: pkg/capabilities/consensus/ocr3/capability_test.go (hunk 17 above)
Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability_test.go:180]: Yeah, its required to be specfied now, a YAML representation would be:
```yaml
aggregation_method: data_feeds_2_0
aggregation_config: {}
```

File: pkg/capabilities/generate_schemas/main.go (hunk 18 above)
Comment (HenryNguyen5) [reply pkg/capabilities/generate_schemas/main.go:11]: Going to remove this file entirely, or rework it so that it regenerates all the test fixtures

File: pkg/capabilities/consensus/ocr3/capability.go (hunk 15 above)
Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability.go:61]: Another reason for public exposure + package level is to have an easy way to generate all of the test fixtures by importing all of the validators in a single file, rather than have a generate script in each package. WDYT?

File: pkg/capabilities/triggers/mercury_trigger_test.go (hunk 20 above)
Comment (HenryNguyen5) [reply pkg/capabilities/triggers/mercury_trigger_test.go:19]: Yeah, this deserves some comments

File: pkg/capabilities/validate.go (hunk 22 above)
Comment (HenryNguyen5) [reply pkg/capabilities/validate.go:30]: We use `ID` and `Description`. I figured it was good to pass in the entire struct since we can store a lot of useful metadata in the schema.

File: pkg/capabilities/validate.go (hunk 23 above)
Comment (HenryNguyen5) [reply pkg/capabilities/validate.go:160]: Sure, could you explain why to me? I'm not knowledgable on when I should choose using `UnwrapTo` rather than just using mapstructure directly

File: pkg/capabilities/validate.go (hunk 24 above)
Comment (HenryNguyen5) [reply pkg/capabilities/validate.go:172]: Would I just deref all of the args to it then?

File: pkg/capabilities/capabilities.go [hunk 25]
```diff
@@ -106,6 +106,20 @@ type CallbackExecutable interface {
...
+	// This schema is used within registration calls, which is when the capability instance is created.
+	ConfigSchema() (string, error)
+	// InputSchema returns the JSON schema for the inputs of the capability.
+	// This schema is used within execution calls, which is when the capability instance is being invoked
+	InputSchema() (string, error)
+	// OutputSchema returns the JSON schema for the output of the capability.
+	// This schema is used to describe the return value of an execution call.
+	OutputSchema() (string, error)
+	// Schema is the combination of all of the above schemas. It is used to describe the capability as a whole.
+	Schema() (string, error)
```
[the remark this answers is not in the document]
Comment (HenryNguyen5) [reply pkg/capabilities/capabilities.go:120]: I can make it just `Schema`, then all of the other methods will be used just for internal validation inside capabilities.

File: pkg/capabilities/consensus/ocr3/capability.go (hunk 14 above)
Comment (cedric-cordenier) [reply pkg/capabilities/consensus/ocr3/capability.go:60]: No, not necessarily -- but in that case I think you don't need line 59, so I'd prefer we pick one and remove the other

File: pkg/capabilities/validate.go (hunk 24 above)
Comment (cedric-cordenier) [reply pkg/capabilities/validate.go:172]: Yep, at the point at which this is called I'd expect you to be guaranteed of having a reflector

File: pkg/capabilities/consensus/ocr3/capability.go (hunk 15 above)
Comment (cedric-cordenier) [reply pkg/capabilities/consensus/ocr3/capability.go:61]: Yeah if you want to test things in isolation I think it's fine to leave it at the package level -- still not sure it needs to be public though? In the end we embed the validator into the capability, and I think it's nice to ensure that we only access it via there.

Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability.go:61]: The value I see exposing it publically would be to generate all test fixtures without having to create an instance of the capability first.

Maybe this isn't a big deal though, I was just thinking it was extra steps if some capabilities are only creatable via a `NewCapability` type function that didn't allow you to pass in empty args.

File: pkg/capabilities/consensus/ocr3/capability.go (hunk 14 above)
Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability.go:60]: I'm confused since capabilityIface contains completely separate methods (used to implement a reporting plugin) whereas ConsensusCapability is to make sure that this struct really is a capability.

I removed the line I added for now. I'd love to understand why we don't enforce the intersection of the two interfaces against this struct.

File: pkg/capabilities/consensus/ocr3/capability.go (hunk 15 above)
Comment (cedric-cordenier) [reply pkg/capabilities/consensus/ocr3/capability.go]: Hmmm... fair point -- you can ignore this then 👍

**Review by cedric-cordenier — DISMISSED** [review]

File: pkg/values/list.go [hunk 26]
```diff
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"reflect"
 
+
```
Comment (cedric-cordenier) [inline pkg/values/list.go:7]: Could you remove this additional line, both here and in map.go @HenryNguyen5 ? Otherwise LGTM

**Review by bolekk — APPROVED** [review]
